pkg/build/strategies/sti/postexecutorstep_test.go

```diff
@@ -0,0 +1,346 @@
+			t.Errorf("should commit container with User: %q, but commited with %q", expectedImageUser, commitOpts.User)
+		}
+
+		if !reflect.DeepEqual(commitOpts.Labels, expectedLabels) {
+			t.Errorf("should commit container with Labels: %v, but commited with %v", expectedLabels, commitOpts.Labels)
+		}
+	}
+}
+
+func TestCommitRuntimeImageStep(t *testing.T) {
+	expectedImageCmd := "/run"
+	expectedEnv := []string{"BUILD_LOGLEVEL"}
+	expectedContainerID := "container-yyyy"
+	expectedImageID := "image-xxx"
+	expectedImageTag := "v1"
+	expectedImageUser := "jboss"
+
+	displayName := "MyApp"
+	description := "My Application is awesome!"
+
+	baseImageLabels := make(map[string]string)
+	baseImageLabels["vendor"] = "CentOS"
+
+	expectedLabels := make(map[string]string)
+	expectedLabels["io.k8s.description"] = description
+	expectedLabels["io.k8s.display-name"] = displayName
+	expectedLabels["vendor"] = "CentOS"
+
+	builder := newFakeBaseSTI()
+	builder.config.DisplayName = displayName
+	builder.config.Description = description
+	builder.config.Tag = expectedImageTag
+	builder.env = expectedEnv
+
+	docker := builder.docker.(*docker.FakeDocker)
+	docker.CommitContainerResult = expectedImageID
+	docker.GetImageUserResult = expectedImageUser
+	docker.Labels = baseImageLabels
+
+	step := &commitRuntimeImageStep{builder: builder, docker: docker}
+
+	ctx := &postExecutorStepContext{
+		containerID: expectedContainerID,
+	}
+
+	if err := step.execute(ctx); err != nil {
+		t.Fatalf("should exit without error, but it returned %v", err)
+	}
+
+	if ctx.imageID != expectedImageID {
+		t.Errorf("should set ImageID field to %q but it's %q", expectedImageID, ctx.imageID)
+	}
+
+	commitOpts := docker.CommitContainerOpts
+
+	if len(commitOpts.Command) != 1 {
+		t.Errorf("should commit container with Command: %q, but commited with %q", expectedImageCmd, commitOpts.Command)
+
+	} else if commitOpts.Command[0] != expectedImageCmd {
+		t.Errorf("should commit container with Command: %q, but commited with %q", expectedImageCmd, commitOpts.Command[0])
+	}
+
+	if !reflect.DeepEqual(commitOpts.Env, expectedEnv) {
+		t.Errorf("should commit container with Env: %v, but commited with %v", expectedEnv, commitOpts.Env)
+	}
+
+	if commitOpts.ContainerID != expectedContainerID {
+		t.Errorf("should commit container with ContainerID: %q, but commited with %q", expectedContainerID, commitOpts.ContainerID)
+	}
+
+	if commitOpts.Repository != expectedImageTag {
+		t.Errorf("should commit container with Repository: %q, but commited with %q", expectedImageTag, commitOpts.Repository)
+	}
+
+	if commitOpts.User != expectedImageUser {
+		t.Errorf("should commit container with User: %q, but commited with %q", expectedImageUser, commitOpts.User)
+	}
+
+	if !reflect.DeepEqual(commitOpts.Labels, expectedLabels) {
+		t.Errorf("should commit container with Labels: %v, but commited with %v", expectedLabels, commitOpts.Labels)
+	}
+}
+
+func TestDownloadFilesFromBuilderImageStep(t *testing.T) {
+	// FIXME
```

Comment:
I hope to that :) I can remove these templates but I believe that it's still good to have tests for these steps. In other hand, writing such tests is hard and I don't have time for them now. I decided to leave here these "reminders".

Do you think that it's really bad?